codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -454,8 +455,12 @@ private void doGracefulShutdown(ChannelHandlerContext ctx, ChannelFuture future,
             future.addListener(new ClosingChannelFutureListener(ctx, promise));
         } else {
             // If there are active streams we should wait until they are all closed before closing the connection.
-            closeListener = new ClosingChannelFutureListener(ctx, promise,
-                                                             gracefulShutdownTimeoutMillis, MILLISECONDS);
+            if (gracefulShutdownTimeoutMillis == -1) {
```

Comment:
nit: we could also just use `< 0` here. at this point we don't care if it is a specific token value (e.g. `-1`) or some other negative number.